lib/index.ts

```diff
@@ -211,33 +288,64 @@ export const convertCircuitJsonToLbrn = (
           continue
         }
 
-        let union = netGeoms[0]!
-        if (union instanceof Box) {
-          union = new Polygon(union)
-        }
-        for (const geom of netGeoms.slice(1)) {
-          if (geom instanceof Polygon) {
-            union = BooleanOperations.unify(union, geom)
-          } else if (geom instanceof Box) {
-            union = BooleanOperations.unify(union, new Polygon(geom))
+        try {
+          let union = netGeoms[0]!
+          if (union instanceof Box) {
+            union = new Polygon(union)
+          }
+          for (const geom of netGeoms.slice(1)) {
+            if (geom instanceof Polygon) {
+              union = BooleanOperations.unify(union, geom)
+            } else if (geom instanceof Box) {
+              union = BooleanOperations.unify(union, new Polygon(geom))
+            }
           }
-        }
 
-        for (const island of union.splitToIslands()) {
-          // Convert the polygon to verts and prims
-          const { verts, prims } = polygonToShapePathData(island)
-
-          project.children.push(
-            new ShapePath({
-              cutIndex: bottomCopperCutSetting.index,
-              verts,
-              prims,
-              isClosed: false,
-            }),
+          for (const island of union.splitToIslands()) {
+            // Convert the polygon to verts and prims
+            const { verts, prims } = polygonToShapePathData(island)
+
+            project.children.push(
+              new ShapePath({
+                cutIndex: bottomCopperCutSetting.index,
+                verts,
+                prims,
+                isClosed: false,
+              }),
+            )
+          }
+        } catch (error) {
+          console.warn(
+            `Failed to union geometries for net ${net} on bottom layer:`,
+            error,
           )
+          // Output individual geometries if union fails
+          for (const geom of netGeoms) {
+            const poly = geom instanceof Box ? new Polygon(geom) : geom
+            const { verts, prims } = polygonToShapePathData(poly)
+            project.children.push(
+              new ShapePath({
+                cutIndex: bottomCopperCutSetting.index,
+                verts,
+                prims,
+                isClosed: false,
+              }),
+            )
+          }
         }
       }
     }
   }
+
+  // Generate trace margin clearance areas (outerTraceUnion - innerTraceUnion)
+  if (traceMargin > 0 && includeCopper) {
+    if (includeLayers.includes("top")) {
+      generateTraceMarginClearance(ctx, "top")
+    }
+    if (includeLayers.includes("bottom")) {
+      generateTraceMarginClearance(ctx, "bottom")
+    }
+  }
+
   return project
```

Comment:
there is still a ton of top level pollution here, also what does "generateTraceMarginClearance" mean? That doesn't make sense. `ctx` is always the SECOND parameter when used, and it's not obvious what the parameters are anyway, so use named parameters.